src/matrix/kaldi-matrix.h

```diff
@@ -574,6 +574,9 @@ class MatrixBase {
   void SymPosSemiDefEig(VectorBase<Real> *s, MatrixBase<Real> *P,
                         Real check_thresh = 0.001);
 
+#if defined(__NVCC__) && defined(_MSC_VER)
```

Comment:
Can you explain the purpose of this? I do not understand why it is here at all.

Reply:
This is needed for successfully compilation CUDA related code on windows. They are absolutely relevant with this PR. There are some weird issue with template friend function in a class template in Windows version of nvcc. This is simple an ugly walkaround.